discord/client.py

```diff
@@ -682,7 +682,8 @@ def activity(self, value: Optional[ActivityTypes]) -> None:
         if value is None:
             self._connection._activity = None
         elif isinstance(value, BaseActivity):
-            self._connection._activity = value.to_dict()
+            # ConnectionState._activity is typehinted as ActivityPayload, we're passing Dict[str, Any]
+            self._connection._activity = value.to_dict() # type: ignore
```

Comment:
I'm not sure why this is needed since the return value of `BaseActivity.to_dict()` is typed as `ActivityPayload`

Reply:
I didn't see that `value` is typed as `Optional[ActivityTypes]`, which is why `to_dict()` doesn't return `ActivityPayload`. This should stay as `type: ignore`